src/App.tsx

```diff
@@ -17,18 +71,32 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                query={query}
+                onSetQuery={handleSetQuery}
+                handleSetStatus={handleSetStatus}
+                status={status}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {todosAreLoading
+              && <Loader />}
+              <TodoList
+                todos={visiableTodos}
+                onTodoSelected={onTodoSelected}
+                selectedTodo={selectedTodo}
+              />
```

Comment:
I think will be better to render TodoList only if you already have false in todosAreLoading.
now you have an empty table while getting todos from server.